internal/reloader/reloader_test.go

```diff
@@ -2,15 +2,277 @@ package reloader
 
 import (
 	"bytes"
-	"log"
+	"context"
+	"errors"
 	"os"
+	"path"
 	"path/filepath"
 	"testing"
 	"time"
 
+	"github.com/fsnotify/fsnotify"
 	"github.com/stretchr/testify/assert"
+	"github.com/supabase/auth/internal/conf"
+	"golang.org/x/sync/errgroup"
 )
 
+func TestWatch(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	dir, cleanup := helpTestDir(t)
+	defer cleanup()
+
+	// test broken watcher
+	{
+		sentinelErr := errors.New("sentinel")
+		rr := mockReloadRecorder()
+		rl := NewReloader(dir)
+		rl.watchFn = func() (watcher, error) { return nil, sentinelErr }
+
+		err := rl.Watch(ctx, rr.configFn)
+		if exp, got := sentinelErr, err; exp != got {
+			t.Fatalf("exp err %v; got %v", exp, got)
+		}
+	}
+
+	// test watch invalid dir
+	{
+		doneCtx, doneCancel := context.WithCancel(ctx)
+		doneCancel()
+
+		rr := mockReloadRecorder()
+		rl := NewReloader(path.Join(dir, "__not_found__"))
+		err := rl.Watch(doneCtx, rr.configFn)
+		if exp, got := context.Canceled, err; exp != got {
+			t.Fatalf("exp err %v; got %v", exp, got)
+		}
+	}
+
+	// test watch error chan closed
+	{
+		rr := mockReloadRecorder()
+		wr := newMockWatcher(nil)
+		wr.errorCh <- errors.New("sentinel")
+		close(wr.errorCh)
+
+		rl := NewReloader(dir)
+		rl.watchFn = func() (watcher, error) { return wr, nil }
+
+		err := rl.Watch(ctx, rr.configFn)
+		if err == nil {
+			t.Fatal("exp non-nil err")
+		}
+		if exp, got := "fsnotify error channel was closed", err.Error(); exp != got {
+			t.Fatalf("exp err %v; got %v", exp, got)
+		}
```

Comment:
I see `assert` is used below. Not a big deal but would be easier if this was just using assert

Reply:
Fixed